Presentation/Executors/ExecutorsForBooks/BookHandlerExecutor.cs

```diff
@@ -95,15 +95,44 @@ private async Task DeleteBook()
         AppPartialsRenderer.RenderHeader();
         ConsoleMessageRenderer.RenderIndicatorMessage("Delete a book");
 
-        var bookId = AnsiConsole.Ask<Guid>("Enter the book ID: ");
-        var wasConfirmed = AnsiConsole.Confirm("Are you sure you want to delete this book?");
+        // var bookId = AnsiConsole.Ask<long>("Enter the book ID: ");
+        // var wasConfirmed = AnsiConsole.Confirm("Are you sure you want to delete this book?");
+        var books = (await _bookRepository.GetAll()).ToArray();
+
+        if (books == null || books.Length == 0)
+        {
+            AnsiConsole.MarkupLine("[red]There are no books available for deletion.[/]");
+            return;
+        }
+
+        var bookToDelete = AnsiConsole.Prompt(
```

Comment:
The selection is fine, do the same for the other methods such as editing books, reports.
![Screenshot from 2024-09-01 10-35-02](https://github.com/user-attachments/assets/03ecb127-e1a6-445e-9ee9-701b5f7e86ff)

![Screenshot from 2024-09-01 10-41-31](https://github.com/user-attachments/assets/936f3efb-660e-4556-9f7d-9e9f7952de75)



To avoid code repetition, you can separate that logic into another class.

Reply:
done